server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DatabasePermissionController.java

```diff
@@ -0,0 +1,103 @@
+public class DatabasePermissionController {
+
+    @Autowired
+    private DatabasePermissionService service;
+
+    @ApiOperation(value = "listDatabasePermissions", notes = "List database permissions")
+    @RequestMapping(value = "/", method = RequestMethod.GET)
+    public PaginatedResponse<DatabasePermissionModel> list(@PathVariable Long projectId,
+            @RequestParam(name = "user") Long userId,
+            @RequestParam(name = "database", required = false) String fuzzyDatabaseName,
+            @RequestParam(name = "datasource", required = false) String fuzzyDatasourceName,
+            @RequestParam(name = "environment", required = false) List<Long> environmentIds,
+            @RequestParam(name = "permissionType", required = false) List<DatabasePermissionType> permissionTypes,
+            @RequestParam(name = "sourceType", required = false) List<PermissionSourceType> sourceTypes,
+            @RequestParam(name = "expired", required = false) List<Boolean> expiredList,
+            @RequestParam(name = "expireSoon", required = false) Boolean expireSoon,
+            @PageableDefault(size = Integer.MAX_VALUE, sort = {"id"}, direction = Direction.DESC) Pageable pageable) {
+        Boolean expired = CollectionUtils.size(expiredList) == 1 ? expiredList.get(0) : null;
+        QueryDatabasePermissionParams params = QueryDatabasePermissionParams.builder()
+                .userId(userId)
+                .fuzzyDatabaseName(fuzzyDatabaseName)
+                .fuzzyDatasourceName(fuzzyDatasourceName)
+                .environmentIds(environmentIds)
+                .permissionTypes(permissionTypes)
+                .sourceTypes(sourceTypes)
+                .expired(expired)
+                .expiredSoon(expireSoon)
+                .build();
+        return Responses.paginated(service.list(projectId, params, pageable));
+    }
+
+    @ApiOperation(value = "createDatabasePermissions", notes = "Create database permissions")
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
+    public ListResponse<DatabasePermissionModel> create(@PathVariable Long projectId,
+            @RequestBody CreateDatabasePermissionReq req) {
+        return Responses.list(service.create(projectId, req));
+    }
+
+    @ApiOperation(value = "reclaimDatabasePermission", notes = "Reclaim database permission")
+    @RequestMapping(value = "/reclaim", method = RequestMethod.DELETE)
+    public SuccessResponse<DatabasePermissionModel> reclaim(@PathVariable Long projectId,
+            @RequestParam(name = "id") Long id) {
+        return Responses.success(service.reclaim(projectId, id));
+    }
+
+    @ApiOperation(value = "releaseDatabasePermission", notes = "Release database permission")
+    @RequestMapping(value = "/release", method = RequestMethod.DELETE)
+    public SuccessResponse<DatabasePermissionModel> release(@PathVariable Long projectId,
+            @RequestParam(name = "id") Long id) {
```

Comment:
What's the different between `reclaim` and `release`?

Reply:
The `release` is used for common user and the `revoke` is used for administrator. The authentication is different and I prefer to use two different API to represent them.

Reply:
Do you mean `release` means common users give up their permissions or revoke other users' permissions?

Reply:
yes

Reply:
Anyway, `release some permission` is not a formal usage.
If you want to express the meaning that it is the users proactive in giving up their permission, you could say relinquish the permission; if you want to express that it is the admin user proactive in revoking other users' permissions, you could just say revoke/withdraw the permission.

Reply:
Okay, use `relinquish`  is more suitable. 👍🏻